File: src/ast/PackedTemplateParameter.java
package ast;

import java.io.StringWriter;
import java.util.List;

public class PackedTemplateParameter extends NodeArray {
    public PackedTemplateParameter(List<BaseNode> nodes) {
        super(nodes, NodeType.PackedTemplateParameter);
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        for (BaseNode node : Nodes) {
            node.PrintLeft(writer);
        }
    }

    @Override
    public void PrintRight(StringWriter writer) {
        for (BaseNode node : Nodes) {
            node.PrintLeft(writer);
        }
    }

    @Override
    public boolean HasRightPart() {
        for (BaseNode node : Nodes) {
            if (node.HasRightPart()) {
                return true;
            }
        }

        return false;
    }
}

File: src/de/fearlesstobi/demangler/ast/PackedTemplateParameter.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class PackedTemplateParameter(nodes: List<BaseNode?>?) : NodeArray(nodes, NodeType.PackedTemplateParameter) {
    override fun printLeft(writer: StringWriter) {
        for (node in nodes!!) {
            node!!.printLeft(writer)
        }
    }

    public override fun printRight(writer: StringWriter) {
        for (node in nodes!!) {
            node!!.printLeft(writer)
        }
    }

    public override fun hasRightPart(): Boolean {
        for (node in nodes!!) {
            if (node!!.hasRightPart()) {
                return true
            }
        }
        return false
    }
}